client/src/templates/Challenges/classic/editor.css

```diff
@@ -55,6 +55,10 @@ textarea.inputarea {
   max-width: unset !important;
 }
 
+code {
+  white-space: pre;
+}
```

Comment:
I think all the other problems stem from this selector being too broad. The original issue is with code block inside the challenge description, but this affects all code elements.

If you narrow this selector, it should be possible to revert all the other changes.

Reply:
You are right. Thanks for the pointer.